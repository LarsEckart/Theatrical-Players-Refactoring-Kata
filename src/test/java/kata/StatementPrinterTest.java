package kata;

import com.github.larseckart.tcr.TestCommitRevertMainExtension;
import org.approvaltests.Approvals;
import org.approvaltests.core.Options;
import org.approvaltests.reporters.Junit5Reporter;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.util.List;
import java.util.Map;

import static org.assertj.core.api.Assertions.assertThat;

@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
@ExtendWith(TestCommitRevertMainExtension.class)
class StatementPrinterTest {

    @Test
    void golden_master() {
        var performances = List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("hamlet", 29),
                new Performance("hamlet", 31),
                new Performance("hamlet", 15),
                new Performance("as-like", 35),
                new Performance("as-like", 12),
                new Performance("as-like", 20),
                new Performance("othello", 40));
        var plays = Map.of(
                "hamlet", new Play("Hamlet", "tragedy"),
                "as-like", new Play("As You Like It", "comedy"),
                "othello", new Play("Othello", "tragedy"));
        Invoice invoice = new Invoice("BigCo", performances);

        String statement = new StatementPrinter(plays).print(invoice);

        Approvals.verify(statement, new Options(new Junit5Reporter()));
    }

    @Test
    void golden_master_html() {
        var performances = List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("hamlet", 29),
                new Performance("hamlet", 31),
                new Performance("hamlet", 15),
                new Performance("as-like", 35),
                new Performance("as-like", 12),
                new Performance("as-like", 20),
                new Performance("othello", 40));
        var plays = Map.of(
                "hamlet", new Play("Hamlet", "tragedy"),
                "as-like", new Play("As You Like It", "comedy"),
                "othello", new Play("Othello", "tragedy"));
        Invoice invoice = new Invoice("BigCo", performances);

        String statement = new HtmlStatementPrinter(plays).print(invoice);

        Approvals.verifyXml(statement, new Options(new Junit5Reporter()));
    }
}
