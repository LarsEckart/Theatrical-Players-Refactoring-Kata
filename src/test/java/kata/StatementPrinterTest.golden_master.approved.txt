Statement for BigCo
  Hamlet: $650.00 (55 seats)
  Hamlet: $400.00 (30 seats)
  Hamlet: $400.00 (29 seats)
  Hamlet: $410.00 (31 seats)
  Hamlet: $400.00 (15 seats)
  As You Like It: $580.00 (35 seats)
  As You Like It: $336.00 (12 seats)
  Othello: $500.00 (40 seats)
Amount owed is $3,676.00
You earned 50 credits
